src/components/Divider.tsx

```diff
@@ -1,70 +1,141 @@
+    const innerThoughtElement = document.querySelector(
+      `[aria-label="tree-node"][data-id="${thought.id}"] [aria-label="thought"]`,
+    )
+    return innerThoughtElement?.getBoundingClientRect().width ?? 0
+  })
+}
 
 /** A custom horizontal rule. */
 const Divider = ({ path, cssRaw }: { path: Path; cssRaw?: SystemStyleObject }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const dividerRef = useRef<HTMLDivElement>(null)
+  const [dividerWidth, setDividerWidth] = useState<number>(DIVIDER_MIN_WIDTH)
+  const { isOnlyChild, isTableView, children, thoughtsAtSameDepth } = useDividerData(path)
+  const editingThoughtId = useSelector((state: State) => state.cursor && head(state.cursor))
+  const editingValueUntrimmed = editingValueUntrimmedStore.useSelector(editingValue => editingValue)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
-  const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.querySelector('[data-divider=true]')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
-    }
-  }
+  useEffect(() => {
+    /** Calculates and updates the Divider's width based on sibling thought widths. */
+    const updateDividerWidth = () => {
+      if (!dividerRef.current) return
 
-  useEffect(setStyle)
+      if (isOnlyChild && !isTableView) {
+        setDividerWidth(DIVIDER_MIN_WIDTH)
+        return
+      }
 
-  const isTableView = useSelector(state => {
-    const parentId = head(rootedParentOf(state, path))
-    return attributeEquals(state, parentId, '=view', 'Table')
-  })
+      let widths: number[] = []
+      if (isTableView && isOnlyChild) {
+        if (thoughtsAtSameDepth.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(thoughtsAtSameDepth)
+      } else {
+        const siblingThoughts = children.filter(child => !isDivider(child.value))
+        if (siblingThoughts.length === 0) {
+          setDividerWidth(DIVIDER_MIN_WIDTH)
+          return
+        }
+        widths = getThoughtWidths(siblingThoughts)
+      }
+
+      setDividerWidth(Math.round(Math.max(...widths, DIVIDER_MIN_WIDTH)))
```

Comment:
Can we omit `Math.round`? I believe browsers handle subpixel rendering just fine.